### PR TITLE
[Feature Store] Add stream source to ingestion service

### PR DESCRIPTION
**Pull request by dinal**

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -469,19 +481,43 @@ def deploy_ingestion_service(
...
+            shard_count=source.shards,
+            retention_period_hours=source.retention_in_hours,
+            raise_for_status=v3io.dataplane.RaiseForStatus.never,
+        )
+        function.add_v3io_stream_trigger(
+            source.path, source.name, source.group, source.seek_to, source.shards
+        )
+    else:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            f"Source type {type(source)} is not supported with ingestion_service yet"
```
Comment (gtopper) [inline mlrun/feature_store/api.py:517]: ingestion_service -> ingestion service

**Review by gtopper — APPROVED** [review]

**Review by Hedingber — APPROVED** [review]

File: mlrun/datastore/sources.py [hunk 2]
```diff
@@ -277,6 +277,10 @@ class OnlineSource(BaseSourceDriver):
         "time_field",
         "online",
         "workers",
+        "group",
```
Comment (yaronha) [inline mlrun/datastore/sources.py:280]: @dinal all those new fields need to go under the `attributes` dict

File: mlrun/datastore/sources.py [hunk 3]
```diff
@@ -312,11 +316,41 @@ class HttpSource(OnlineSource):
...
+        self,
+        name="stream",
+        group="serving",
+        seek_to="earliest",
+        shards=1,
+        retention_in_hours=24,
+        **kwargs,
+    ):
+        super().__init__(name, **kwargs)
+        self.group = group
```
Comment (yaronha) [inline mlrun/datastore/sources.py:342]: @dinal same, set the values to the `attributes` dict (the UI sets the values in the attributes based on the kind)

File: mlrun/feature_store/api.py [hunk 4]
```diff
@@ -449,6 +453,14 @@ def deploy_ingestion_service(
         featureset = get_feature_set_by_uri(featureset)
 
     run_config = run_config.copy() if run_config else RunConfig()
+    if isinstance(source, StreamSource) and (
+        source.path is None or source.path == "None"
```
Comment (yaronha) [inline mlrun/feature_store/api.py:457]: why "None" ?

File: mlrun/feature_store/api.py [hunk 5]
```diff
@@ -469,19 +481,44 @@ def deploy_ingestion_service(
...
     )
     function.verbose = function.verbose or verbose
+    add_source_trigger(source, function)
+
     if run_config.local:
         return function.to_mock_server(namespace=get_caller_globals())
     return function.deploy()
 
 
+def add_source_trigger(source, function):
```
Comment (yaronha) [inline mlrun/feature_store/api.py:497]: i guess this func belongs on a different file (sources or ingestion) not in the API

File: mlrun/model.py [hunk 6]
```diff
@@ -1014,7 +1014,7 @@ def __init__(
     ):
 
         self.name = name
-        self.path = str(path)
+        self.path = str(path) if path is not None else None
```
Comment (yaronha) [inline mlrun/model.py:1017]: isn't the opposite of not None = None?

Comment (dinal) [reply mlrun/model.py:1017]: if path is not None
   self.path = str(path)
else
   self.path = None

Comment (yaronha) [reply mlrun/model.py:1017]: ok

**Review by Hedingber — APPROVED** [review]

**Pull request merged**